src/main/kotlin/org/jitsi/rtp/rtcp/CompoundRtcpPacket.kt

```diff
@@ -16,24 +16,49 @@
 
 package org.jitsi.rtp.rtcp
 
-class CompoundRtcpPacket(
-    buffer: ByteArray,
-    offset: Int,
-    length: Int
-) : RtcpPacket(buffer, offset, length) {
-
-    val packets: List<RtcpPacket> = run {
-        var bytesRemaining = length
-        var currOffset = offset
-        val rtcpPackets = mutableListOf<RtcpPacket>()
-        while (bytesRemaining >= RtcpHeader.SIZE_BYTES) {
-            val rtcpPacket = parse(buffer, currOffset)
-            rtcpPackets.add(rtcpPacket)
-            currOffset += rtcpPacket.length
-            bytesRemaining -= rtcpPacket.length
+import org.jitsi.rtp.util.BufferPool
+
+class CompoundRtcpPacket private constructor(
```

Comment:
The other RTCP classes use a builder helper, which I think would allow this constructor to remain as it was and replace the need for `Holder`.  Another scheme people use is to put an invoke  operator in the companion object, so it looks just like a constructor but can do some work before hand.  E.g.:
```
class CompoundRtcpPacket(
  buffer: ByteArray,
  offset: Int,
  length: Int
) : RtcpPacket(buffer, offset, length) {
  ...
  companion object {
    operator fun invoke(vararg packets: RtcpPacket) {
      val totalLength = packets.map { it.length }.sum()
      val buf = BufferPool.getArray(totalLength + BYTES_TO_LEAVE_AT_END_OF_PACKET)

      var off = 0
      packets.forEach {
        System.arraycopy(it.buffer, it.offset, buf, off, it.length)
        off += it.length
      }
      return CompoundRtcpPacket(buf, 0, totalLength)  
    }
  }
}
```
That companion object function would be called as `val compound = CompoundRtcpPacket(rtcpPacket1, rtcpPacket2, rtcpPacket3)`